src/modules/launcher/Plugins/Microsoft.Plugin.Uri/UriHelper/ExtendedUriParser.cs

```diff
@@ -59,17 +64,36 @@ public bool TryParse(string input, out System.Uri result, out bool isWebUri)
                 {
                     urlBuilder.Scheme = System.Uri.UriSchemeHttp;
                     isWebUri = true;
+                    secondResult = null;
+                }
+                else if (Regex.IsMatch(input, isDomainPortRegex) ||
+                         Regex.IsMatch(input, isIPv6PortRegex))
+                {
+                    var secondUrlBuilder = urlBuilder;
```

Comment:
Do we need it really. Can't we reuse the uri value from first builder? This might save time.

Reply:
This is mainly because UriBuilder already has a complete parsing system. Additionally the secondResult is returned as a System.Uri, so it will have to be a second UriBuilder anyway.

Potential change for this is to return a Uri String instead for the second result, it shouldn't be too difficult. Let me know what you think.

Reply:
then keep the code we have.